pkg/storage/engine.go

```diff
@@ -72,6 +72,15 @@ type SimpleMVCCIterator interface {
 	// UnsafeValue returns the same value as Value, but the memory is
 	// invalidated on the next call to {Next,NextKey,Prev,SeekGE,SeekLT,Close}.
 	UnsafeValue() []byte
+	// HasPointAndRange returns whether the current iterator position has a point
+	// key and/or a range key. Range keys are only emitted when requested via
+	// IterOptions.KeyTypes.
+	HasPointAndRange() (bool, bool)
+	// RangeBounds returns the range bounds for the current range key, if any.
+	RangeBounds() (roachpb.Key, roachpb.Key)
```

Comment:
I was trying to get my head around this change and I think changes to how range keys are iterated deserves additional comments. While it is very straightforward how we iterate values, it is not so with range keys in terms of key bounds. Is RangeBounds a union of all RangeKeys() overlapping at current timestamp?